resource/opentelemetry-resource-detector-azure/src/opentelemetry/resource/detector/azure/vm.py

```diff
@@ -68,7 +68,7 @@ def get_azure_vm_metadata(self):  # pylint: disable=no-self-use
         request = Request(_AZURE_VM_METADATA_ENDPOINT)
         request.add_header("Metadata", "True")
         try:
-            with urlopen(request, timeout=10) as response:
+            with urlopen(request, timeout=4) as response:
```

Comment:
Changing timeout to 4s is probably a short term solution but should we find out what the actual interaction is between this and concurrent.futures?

Reply:
Done.